polycommit/src/sonic_pc/gadgets/verifier_key/prepared_verifier_key.rs

```diff
@@ -189,73 +113,10 @@ where
     >(
-        mut cs: CS,
-        value_gen: Fn,
+        _cs: CS,
+        _value_gen: Fn,
     ) -> Result<Self, SynthesisError> {
-        let obj = value_gen()?.borrow().clone();
-
-        let mut prepared_g = Vec::<PG::G1Gadget>::with_capacity(obj.prepared_vk.prepared_g.len());
-        for (i, g) in obj.prepared_vk.prepared_g.iter().enumerate() {
-            prepared_g.push(<PG::G1Gadget as AllocGadget<
-                <TargetCurve as PairingEngine>::G1Projective,
-                <BaseCurve as PairingEngine>::Fr,
-            >>::alloc_constant(cs.ns(|| format!("g_{}", i)), || {
-                Ok(g.into_projective())
-            })?);
-        }
-
-        let mut prepared_gamma_g = Vec::<PG::G1Gadget>::with_capacity(obj.prepared_vk.prepared_gamma_g.len());
-        for (i, gamma_g) in obj.prepared_vk.prepared_gamma_g.iter().enumerate() {
-            prepared_gamma_g.push(<PG::G1Gadget as AllocGadget<
-                <TargetCurve as PairingEngine>::G1Projective,
-                <BaseCurve as PairingEngine>::Fr,
-            >>::alloc_constant(
-                cs.ns(|| format!("gamma_g_{}", i)), || Ok(gamma_g.into_projective())
-            )?);
-        }
-
-        let prepared_h = PG::G2PreparedGadget::alloc(cs.ns(|| "prepared_h"), || Ok(&obj.prepared_vk.prepared_h))?;
-        let prepared_beta_h =
-            PG::G2PreparedGadget::alloc(cs.ns(|| "prepared_beta_h"), || Ok(&obj.prepared_vk.prepared_beta_h))?;
-
-        let degree_bounds_and_prepared_neg_powers_of_h = if obj.degree_bounds_and_prepared_neg_powers_of_h.is_some() {
-            let mut res = Vec::<(usize, FpGadget<<BaseCurve as PairingEngine>::Fr>, PG::G2PreparedGadget)>::new();
-
-            for (i, (d, shift_power_elem)) in obj
-                .degree_bounds_and_prepared_neg_powers_of_h
-                .as_ref()
-                .unwrap()
-                .iter()
-                .enumerate()
-            {
-                let gadget = <PG::G2PreparedGadget as AllocGadget<
-                    <TargetCurve::G2Affine as PairingCurve>::Prepared,
-                    <BaseCurve as PairingEngine>::Fr,
-                >>::alloc_constant(
-                    cs.ns(|| format!("alloc_constant_gadget_{}", i)),
-                    || Ok(shift_power_elem),
-                )?;
-
-                let d_gadget = FpGadget::<<BaseCurve as PairingEngine>::Fr>::alloc_constant(
-                    cs.ns(|| format!("alloc_constant_d_{}", i)),
-                    || Ok(<<BaseCurve as PairingEngine>::Fr as From<u128>>::from(*d as u128)),
-                )?;
-
-                res.push((*d, d_gadget, gadget));
-            }
-            Some(res)
-        } else {
-            None
-        };
-
-        Ok(Self {
-            prepared_g,
-            prepared_gamma_g,
-            prepared_h,
-            prepared_beta_h,
-            degree_bounds_and_prepared_neg_powers_of_h,
-            origin_vk: None,
-        })
```

Comment:
The intended method of constructing a prepared_verifier_key_var is via verifier_key_var.prepare()